server/cluster/cluster.go

```diff
@@ -329,6 +329,19 @@ func (c *RaftCluster) Start(s Server) error {
 		return err
 	}
 	c.checkTSOService()
+	defer func() {
+		// We need to try to stop tso jobs when the cluster is not running.
+		// Ref: https://github.com/tikv/pd/issues/8836
+		if !c.running {
```

Comment:
I think it is better to check `runing` rather than error. Firstly,  we need to call `stopTSOJobsIfNeeded` whenever we return early, regardless of whether error is nil or not. Secondly, it prevents us from modifying the code later to return the error without assigning a value to it.